app/ante/utils_test.go

```diff
@@ -343,6 +344,35 @@ func (suite *AnteTestSuite) CreateTestEIP712MsgSubmitEvidence(from sdk.AccAddres
 	return suite.CreateTestEIP712CosmosTxBuilder(from, priv, chainId, gas, gasAmount, msgEvidence)
 }
 
+func (suite *AnteTestSuite) CreateTestEIP712SubmitProposalV1(from sdk.AccAddress, priv cryptotypes.PrivKey, chainId string, gas uint64, gasAmount sdk.Coins) client.TxBuilder {
+	// Build V1 proposal messages. Must all be same-type, since EIP-712
+	// does not support arrays of variable type.
+	recipient1 := sdk.AccAddress(common.Address{}.Bytes())
+	msgSend1 := types2.NewMsgSend(from, recipient1, sdk.NewCoins(sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewInt(1))))
+
+	recipient2 := sdk.AccAddress(common.HexToAddress(
+		"0x4242424242424242424242424242424242424242",
+	).Bytes())
+	msgSend2 := types2.NewMsgSend(from, recipient2, sdk.NewCoins(sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewInt(1))))
+
+	proposalMsgs := []sdk.Msg{
+		msgSend1,
+		msgSend2,
+	}
+
+	// Build V1 proposal
+	msgProposal, err := govtypes.NewMsgSubmitProposal(
+		proposalMsgs,
```

Comment:
can we use a common proposal type instead (for example, the TextProposal or Param change)?